### PR TITLE
Added `run_tokenizer` method in ` caikit_nlp/modules/text_generation/text_generation_local.py`

### PR DESCRIPTION
**Pull request by m-misiura**
There is a method with `text_generation_local.py` which does not appear to be implemented: https://github.com/caikit/caikit-nlp/blob/56b7e18d543a51e510c51182842b3b684a008bd8/caikit_nlp/modules/text_generation/text_generation_local.py#L593

I have written a method, which takes a string as an input and returns the token count (based on the model tokenizer). The output should follow the data_model: `TokenizationResults`

This method was tested with `tox`; the tests include checking outputs in the case of:

1. an empty string
2. a short sentence
3. a relatively long input

The implemented tests seem to pass

### CODE REVIEW

**Comment by m-misiura** [comment]
Many thanks for highlighting the lack of the sign-off @evaline-ju; I've rebased and added the sign-off accordingly

File: tests/modules/text_generation/test_text_generation_local.py [hunk 1]
```diff
@@ -211,7 +212,26 @@ def test_zero_epoch_case(disable_wip):
...
+    # Edge case: Empty string
+    empty_result = model.run_tokenizer("")
+    assert isinstance(empty_result, TokenizationResults)
+    assert empty_result.token_count == 0
+
+    # Normal case: short sentence
+    short_text = "This is a test sentence."
+    short_result = model.run_tokenizer(short_text)
+    assert isinstance(short_result, TokenizationResults)
+    assert short_result.token_count > 0
```
[the remark this answers is not in the document]
Comment (m-misiura) [reply tests/modules/text_generation/test_text_generation_local.py:231]: Checking for an expected number instead of or in addition to a non-zero condition seems like a good idea to increase the robustness of the test. 

What are your thoughts on re-writing such a test like this:
```
assert short_result.token_count == len(model.model.tokenizer.encode(short_text))
```

File: caikit_nlp/modules/text_generation/text_generation_local.py [hunk 2]
```diff
@@ -590,7 +591,11 @@ def run_tokenizer(
             TokenizationResults
                 The token count
         """
-        raise NotImplementedError("Tokenization not implemented for local")
+        error.type_check("<NLP48137045E>", str, text=text)
+        tokenized_output = self.model.tokenizer(text)
```
[the remark this answers is not in the document]
Comment (m-misiura) [reply caikit_nlp/modules/text_generation/text_generation_local.py:595]: Very interesting question! HF in their [documentation](https://huggingface.co/docs/transformers/en/main_classes/tokenizer#transformers.PreTrainedTokenizer.__call__.return_attention_mask) mention that  _If left to the default, will return the attention mask according to the specific tokenizer’s default, defined by the return_outputs attribute._ 

For consistency and clarity with the other modules, it could be prudent to include `return_attention_mask` explicitly. In such a case, the only potential concern would be whether setting `return_attention_mask` explicitly could result in messing with the default behaviour of the tokeniser, but I think the likelihood of this is low. 

Thus, I can change the method to:
```
def run_tokenizer(
        self,
        text: str,
    ) -> TokenizationResults:
        """Run tokenization task against the model

        Args:
           text: str
                Text to tokenize
        Returns:
            TokenizationResults
                The token count
        """
        error.type_check("<NLP48137045E>", str, text=text)
        tokenized_output = self.model.tokenizer(text, return_attention_mask=True)
        return TokenizationResults(
            token_count=len(tokenized_output["input_ids"]),
        )
```

Let me know how you would like me to proceed

[the remark this answers is not in the document]
Comment (m-misiura) [reply caikit_nlp/modules/text_generation/text_generation_local.py]: I've added `return_attention_mask=True` based on our conversation to ensure consistency with the other modules

File: tests/modules/text_generation/test_text_generation_local.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-misiura) [reply tests/modules/text_generation/test_text_generation_local.py]: Great -- changes to tests have been implemented accordingly and they seem to pass
